Avoid using the ``uuid`` module, to avoid mpi errors on Python 3.8.
